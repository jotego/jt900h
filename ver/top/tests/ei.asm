    ; ei / push SR
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header

    ld xsp, stack
    ei 1
    push sr
    ld a,(xsp+1)
    cp a,0x98
    jp ne,bad_end

    ei 4
    incf
    push sr
    ld a,(xsp+1)
    cp a,0xc9
    jp ne, bad_end

    ei 2
    incf
    push sr
    ld a,(xsp+1)
    cp a,0xaa
    jp ne,bad_end

    include finish.inc
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
stack:
    dw 0,0,0,0,0,0,0,0,0,0
    end