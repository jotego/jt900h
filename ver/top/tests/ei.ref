00000098 - 000000c9 - 000000aa - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 0000babe - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000004f
SR = aa42
