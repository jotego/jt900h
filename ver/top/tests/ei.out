00000098 - 000000c9 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 0000babe - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000003d
SR = c942
